### PR TITLE
feature(trigger-matrix): resolve scylla_version per backend (SCT-665)

### PR DESCRIPTION
**Pull request by fruch**
## What

Fixes [SCT-665](https://scylladb.atlassian.net/browse/SCT-665) — tier1 GCE jobs aborting during config validation with `ValueError: GCE image for scylla_version='...' was not found` — and makes the trigger matrix resolve the Scylla build per backend, with strategies for triggers that need every job on the same build.

## Two root causes

**1. GCE couldn't find nightly images at all** (`sdcm/utils/common.py`)

`get_scylla_gce_images_versions()` always filtered on `labels.environment eq 'production'`, but nightly GCE images are labeled `daily` and freshly built release candidates `candidate`. A full version tag — the only way a nightly can be requested — therefore never matched, no matter which backend triggered the job. Checked against the real project: the GCE image for the exact build in the ticket (`2026.3.0~dev-0.20260717.3f9d1c16c708`) existed all along, the filter just hid it.

The environment filter is now dropped when the version names one exact build (debug images excluded by name instead), and the GCE label is built from the parsed tag components, so the `-N` packaging revision AWS carries on release tags (`2025.4.10-0.20260609.99f4121cd8e1-1`) no longer breaks the match either. Open-ended versions (`2025.4`, no version) still resolve to released images only.

**2. One AWS-resolved build was stamped on every job** (`sdcm/utils/trigger_matrix.py`)

`resolve_to_full_version()` resolved `master:latest` once via an AWS AMI lookup and `build_job_parameters()` stamped that tag on every job in the matrix, regardless of backend, region or architecture. Nightly images are built per backend and per architecture, so GCE/Azure/ARM jobs regularly got a build only AWS had published.

## Version resolution strategies

Resolution is now per **backend/region/architecture**, picked by `version_resolution` in the matrix YAML and overridable per run with `--version-resolution` (also exposed as a Jenkins job parameter):

| strategy | every job gets | use it when |
|---|---|---|
| `per-backend` (default) | each backend/region/arch resolves its own latest build | jobs are independent |
| `common` | the newest build published on **every** target — the lowest of their latest builds | results must be comparable across jobs |
| `aws-strict` | the AWS build; jobs whose backend lacks that exact build are **not triggered** | you want the AWS build or nothing |

`common` fails the trigger run when no build is shared; the other two skip only the jobs they can't serve and report them as skipped. `configurations/triggers/perf-regression.yaml` is set to `common` — perf numbers are only comparable when every job measures the same build; the rest keep `per-backend`.

Availability is checked with the same lookups `SCTConfiguration` runs downstream (`get_scylla_ami_versions`, `get_scylla_gce_images_versions`, `azure_utils.get_scylla_images`, `oci_utils.get_scylla_images_by_version`), so a version that passes here can't abort provisioning later.

## Also handled

- **Multi-DC jobs** (`region: '["eu-west-1", "eu-west-2"]'`) — the build must exist in every region the job provisions in, mirroring what the downstream config requires.
- **Architecture** — new `arch: "aarch64"` field on matrix jobs (defaults to `x86_64`), set on the i8g/arm64 perf jobs and the scylla-doctor ARM job. AWS names that architecture `arm64`, which is now mapped correctly for both AMI lookups.
- **Simple versions** — `2025.4` is looked up as `branch-2025.4:latest` instead of being passed verbatim to a branch lookup that always failed.
- Jobs that don't install from an image (rolling-upgrade via `new_scylla_repo`, PGO via `unified_package`) are left out of resolution entirely.
- The AWS reference resolution in `sct.py` is no longer fatal when it fails — per-backend resolution takes over — except under `aws-strict`, where the AWS build *is* the contract.

## Testing

- `unit_tests/trigger_matrix/` — 189 tests, 37 new in `test_version_resolution.py` covering each strategy, GCE label round-tripping, multi-region, arch, and the end-to-end stamping through `trigger_matrix()`. An autouse conftest fixture stubs the cloud lookups: the existing tests started hitting AWS/GCE for real once resolution moved into `trigger_matrix()`.
- `unit_tests/unit/` — 2834 passed.
- Real dry runs against AWS/GCE/Azure: tier1 with `per-backend` and `aws-strict`, perf-regression with `common` across x86_64 + aarch64 targets. Two bugs surfaced there and were fixed before this PR: the ARM AMI arch naming, and the multi-DC JSON region list.

Docs: `docs/testing/trigger-matrix.md`.

[SCT-665]: https://scylladb.atlassian.net/browse/SCT-665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by fruch** [comment]
@claude review

File: sdcm/utils/trigger_matrix.py [hunk 1]
```diff
@@ -658,6 +922,146 @@ def _extract_branch_from_version(scylla_version: str) -> str:
...
+    # published on all targets (i.e. the lowest of the per-backend latest builds).
+    candidates: list[str] = []
+    if resolvable:
+        for target in targets:
+            if target.backend not in VALID_IMAGE_BACKENDS:
+                continue
+            version = _resolve_latest_version_for_backend(original_version, target.backend, target.region, target.arch)
+            if version and version not in candidates:
+                candidates.append(version)
+        candidates.sort(key=_version_build_date, reverse=True)
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/utils/trigger_matrix.py:1310]: Acknowledged, but not changing this — the gap is narrower than it looks, and it fails loudly rather than silently picking a wrong build.

The candidate set is `{latest build per target}`, tried newest-first against every target. So the resolution is already correct whenever the backends' build sets are *aligned*, which is the normal case — including "AWS is a nightly ahead of GCE": AWS's latest is rejected, GCE's latest is present on AWS too, and that becomes the common build. `test_common_picks_the_newest_build_published_everywhere` covers exactly that.

The true answer is the newest build in the intersection of all targets' build sets, which is always `<= min(latest_i)` — and `min(latest_i)` *is* in the candidate set. So the only input that defeats this is one where a backend skipped a build that another backend's latest happens to be, making the intersection's newest element strictly older than every target's latest. In that case `common` raises, naming the targets missing each candidate and pointing at `per-backend` / `aws-strict`. No job is ever handed a build that isn't published for it, which is what SCT-665 was about.

Searching all eligible builds per target would need new plumbing — `_resolve_latest_version_for_backend` returns a single build per backend, and enumerating candidates properly means listing and paginating images across AWS regions, GCE, Azure and OCI, then intersecting. That's worth doing as a follow-up if we ever see the raise in practice, but it isn't a precondition for this PR.

Two smaller notes, since they came up in the other review of this branch:

- The `environment=production` filter drop in `_get_ami_versions` is pre-existing (commit 2fd7bef3f8, Jan 2026) and untouched here — this PR only documents the equivalent behaviour on the GCE side.
- The suggested "missing test for `common` with an empty `reference_version`" isn't reachable: `resolve_versions_for_targets` starts with `reference_version = reference_version or original_version`, so an empty reference collapses to the original version. The empty-candidate raise is already covered by `test_common_raises_when_nothing_resolves`.

For the record, this branch was rebased onto master (101 commits) and re-verified: 286 unit tests green, and live dry-runs against real AWS/GCE images for both new strategies — `tier1` (`per-backend`) resolved 6 targets with the multi-DC job keeping both its regions, and `perf-regression` (`common`) agreed on one build across 6 targets including `aarch64` and `gce`.
